keyvi/include/keyvi/dictionary/dictionary_merger.h

```diff
@@ -197,20 +243,69 @@ class DictionaryMerger final {
 
     TRACE("finished iterating, do final compile.");
 
-    generator->CloseFeeding();
+    generator_->CloseFeeding();
+  }
+
+  void Write(std::ostream& stream) {
+    if (!generator_) {
+      throw merger_exception("not merged yet");
+    }
 
-    generator->SetManifestFromString(manifest_);
-    generator->WriteToFile(filename);
+    generator_->Write(stream);
   }
 
+  void WriteToFile(const std::string& filename) {
+    if (!generator_) {
+      throw merger_exception("not merged yet");
+    }
+    generator_->WriteToFile(filename);
+  }
+
+  const MergeStats& GetStats() const { return stats_; }
+
  private:
+  typename GeneratorAdapter::AdapterPtr generator_;
   bool append_merge_ = false;
   std::vector<fsa::automata_t> dicts_to_merge_;
+  std::vector<std::vector<std::string>> deleted_keys_;
   std::vector<std::string> inputFiles_;
   std::priority_queue<SegmentIterator> segments_pqueue_;
-
-  fsa::internal::IValueStoreWriter::vs_param_t params_;
+  keyvi::util::parameters_t params_;
   std::string manifest_ = std::string();
+  MergeStats stats_;
+
+  /**
+   * Load a file with deleted keys if it exists
+   */
+  std::vector<std::string> TryLoadDeletedKeys(const std::string& filename) {
+    std::vector<std::string> deleted_keys;
+    boost::filesystem::path deleted_keys_file{filename};
+    deleted_keys_file += ".dk";
```

Comment:
Looks like this naming assumption is only for `Index`, maybe it worth to have merger more independent and make this configurable or provided from outside ? (no strong opinion though)

Reply:
yeah, I thought about an explicit parameter in the merger. On the other side you have to create a file called 'keyvifile.kv.dk' - rather unlikely to do this by mistake.

Thoughts?

Reply:
Fair enough.